### ISSUE TITLE
Chat redesign

### ISSUE THREAD

**Issue by Zughy** [minetest/minetest#10372]
## Problem

The upper left corner is not a "friendly" spot for the eye. When people play, they tend to look at the centre of the screen, not at corners, especially not at the upper ones. Noticing messages is kind of hard

## Solutions

Put the chat where the majority of games do - and for a reason -: somewhere on the bottom-ish left. E.g. Minecraft, Fortnite, Xonotic, Apex Legends, S4 League, literally every RPG. Also, there should be a bounding box, in order to avoid this #10364.

Possibly with a little bit of semi-transparent background as I did here with names #10152 

And also the possibility to scroll it back and forth with the mouse wheel would be nice.

## Additional Context

The F10 one can still be kept  to show a full log, idk

**Comment by LoneWolfHT** [comment]
A semi-transparent background will solve all of my problems with the current chat

**Comment by paramat** [comment]
Obstructing the view of the sky/ceiling is less of a problem than obstructing the view of the ground/floor. Perhaps chat is intentionally placed in the least visually obstructive place. If chat log length is increased, obscuring the lower left will be more visually intrusive than obscuring the upper left.

It might have to be optional (and therefore more complex) as it will overlap a position used by many mods for custom HUD elements.
It might overlap wide custom hotbars.

The lower edge of the screen seems like it is for graphical world-interaction stuff like wieldhand, hotbar and custom HUD stuff, the top edge seems more suitable for 'texty' stuff like F5 debug info and chat.
I am suggesting that lower left is a high-priority position and therefore should not be used for chat, because there are higher priorities for that position, and some things only look right being on the lower edge of the screen, while chat is fine at top edge.
If chat goes lower left then custom HUDs will be forced elsewhere, like the top edge of the screen, where they will be in a very unsuitable place and look very wrong.

At first i thought this might be a good change, but thinking further i am beginning to suspect it is best to leave chat where it is.

**Comment by benrob0329** [comment]
Chat as a HUD element isn't a bad idea, different games may use it differently. Casual building games don't need it to be in your face, but a Team Death Match or Capture The Flag game might want it there as it's very important teams stay aware of each other.

**Comment by Zughy** [comment]
I don't agree, Paramat: using HUD elements seems an excuse to me, even infotext is on the left (as for now). Infotexts are way less used than chats, so I wouldn't use what people did in other mods as something to prevent (what I think is) an improvement. For instance, chat is an essential element to any multiplayer server. Also, I didn't mean the very lower left corner, and it actually scares me you didn't understand what I meant as it gives me the idea you haven't really played a lot of games (or even a few). I mean, I am no professional game designer, but  shouldn't be a little bit of gaming experience a prerequisite to be the one with the most contributions to an engine which generates from a very specific game? I'm not a native English speaker as well, but you talked about overlaying hotbars, so I'm pretty sure you meant the corner.

Minecraft chat
![](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fi.ytimg.com%2Fvi%2FPkKFO8NiE6g%2Fmaxresdefault.jpg)

S4 League chat (2009)
![](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Forig00.deviantart.net%2F8f7e%2Ff%2F2014%2F002%2Fe%2Fe%2Fs4_league_event___tour_de_4_joueurs_by_karaokruk-d70k3fy.png)

Apex Legends chat - open
![](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fcdn.eslgaming.com%2Fmisc%2Fmedia%2Fplay%2Fapexlegends%2Fapex-example.png)

**Comment by rubenwardy** [comment]
Whilst I support chat in the bottom left (I do it for my own games), I worry about breaking and causing issues with mods. I'd like to have a new HUD API that makes it much easier to prevent HUD elements conflicting, and also allow full control of the builtin without having to remake everything

**Comment by v-rob** [comment]
There seems to be a lot of [argumentum ad populum](https://en.wikipedia.org/wiki/Argumentum_ad_populum) going on here.  Just because other games do it is not a good reason.  The only reason I see is that it is in a more "friendly" spot, and I find this unconvincing due to the cons.  Here is my opinion (screenshots from Sonic Robo Blast 2, which has an option for chat at both the top of the screen and the bottom left area):

Bottom Left:
![image](https://user-images.githubusercontent.com/31123645/92646072-b859ce00-f29a-11ea-970e-17b48a1c62fd.png)
* Easy to see because of the position, but by the same token obstructs the view much more.
* It has a small box, so it can't hold as much text, wraps much more, and gets tall fast.  This contributes to the view obstruction.
* Likely to conflict with Minetest's current HUD placement

Top:
![image](https://user-images.githubusercontent.com/31123645/92645764-35387800-f29a-11ea-894b-128aa8f6e4b8.png)

* Trickier to see, **but** it has no semi-transparent background.  Adding that would definitely make it more visible.  In fact, early C55 versions of Minetest had a chatbox at the bottom of the screen with such a background*, and it is much more visible than without one, as can be seen.
* It uses the entire width of the screen, so it doesn't get as tall nearly as fast.
* In this example, it conflicts with the HUD, but Minetest's HUD is at the bottom of the screen, so there is no confliction.

So, I would definitely support a semi-transparent background for the current chat, but I oppose a bottom left box.

*This is the C55 screenshot:
![flowing](https://user-images.githubusercontent.com/31123645/92646649-9876da00-f29b-11ea-9e55-53add7017770.png)

**Comment by benrob0329** [comment]
Taking up the width of the screen is not an inherent improvement, as it can make text more difficult to read (your eyes have to move much further, this is why most websites condense text into the middle 50-75% of wide-screen displays).

The screenshot of SRB2 also shows that game to have numerous issues with it's HUD design, namely that its much too large and padded far too much given the small screen real estate it is trying to partition. This is not an argument against the usage of a lower-left chat, especially considering how bad the top-left chat option looks.

Claiming the reference of successful examples to be argumentum ad populum constitues a lot of dismissive arguments for wanted engine changes, there is probably a name for that as a logical fallacy too if I cared to pour through the list of them. Let's not let this devolve into that, though.

While it is true that bottom-left chat is more obstructive, this isn't a binary situation. The amount of obstructiveness can be controlled, so as not to be in the way too much while still being more easily read in the midst of other activities for the sake of busy gameplay where being aware of the conversation is useful.

**Comment by v-rob** [comment]
> The screenshot of SRB2 also shows that game to have numerous issues with it's HUD design...

True, and I only used it as an example because it's the only game I know that supports both types of chat.  Yet the other screenshots shown previously also seem to have large chatboxes that take a significant part of the screen, and while this can be tweaked to minimize chatbox size while keeping text wrapping to a minimum, I don't think it can be done very well.  Even the Minecraft chatbox among those previous screenshots, while the least obstructive of the bunch, would get in the way with lots of comments.

> Taking up the width of the screen is not an inherent improvement, as it can make text more difficult to read...

Maybe so, but some long conversations I've seen in chat make the extra space invaluable.  There are other ways to alleviate this eye strain, like make it slightly less than the size of the screen or increasing the font size.

> Claiming the reference of successful examples to be argumentum ad populum constitues a lot of dismissive arguments for wanted engine changes

I probably shouldn't have mentioned the fallacy, but I just didn't see many reasons for why a bottom left chatbox would be better than what we already have.  The main reason seemed to be just because other games had it, and that doesn't convince me.  I didn't mean to be dismissive, just get more reasons.

Finally, I do support moving the chatbox around as an HUD element, but I don't want to see the bottom left thing become the default unless I can see something that addresses my concerns.

**Comment by Zughy** [comment]
v-rob, we have to distinguish populism from empiricism. I've been playing videogames for ages, so many I forgot most of them. As far as I recall, I don't remember playing games with a bad UX when it comes to chat, but I can definitely say Minetest chat is not practical. So here's the problem: either all the other games are plain wrong and I'm odd (should read "retarded"), or Minetest chat is actually not practical.
Do I really have to look for papers explaining the science behind chat positioning? Don't you all have a wealth of experience to compare what I stated to? MT chat right now looks like a terminal, and even if developers might be intrigued by it, I'd like to remember developers aren't the only ones playing MT. That's why I said the F10 function could be kept, so everyone's happy.
Also: that Sonic chat is way too invasive

**Comment by Kezii** [comment]
> Obstructing the view of the sky/ceiling is less of a problem than obstructing the view of the ground/floor.

It's **literally the point** of the chat to obstruct the view, you have to see if someone calls you or mentions something you are interested in, that's the point of online playing, right now the chat is useless as a chat because people won't see it unless they are procuring themselves an headache by moving their eyes between the crosshair and the (distant) chat every few seconds.

Also, doing what everyone else is doing is not "argumentum ad populum", it is "providing a familiar user experience for new users", and it's one of the most, if not the most important thing you have to consider when you design an UI

Minecraft solves the "screen visibility" problem by hiding older messages

**Comment by v-rob** [comment]
> It's **literally the point** of the chat to obstruct the view

Fine, then I don't want it.  For me, the main point of Minetest is not chatting.  I need to see what I'm doing.  I consider (and have always considered) chat to be an auxiliary feature, and no auxiliary feature should ever obstruct the main purpose.

> Don't you all have a wealth of experience to compare what I stated to?

Yes, and I decided I didn't like it, and I gave my reasons for not liking it.  Look: I have no opposition to making chat a HUD element that can be moved around; I consider it a wonderful feature.  But I don't like chat in the bottom left area, and I want to argue against that specifically (which is all this issue is about, it says nothing about it being a configuration).  Obviously, I'm outnumbered in my opinion, but that won't prevent me from stating it.

**Comment by LoneWolfHT** [comment]
I also want the chat to stay where it is. It gets in the way where it is in MC.

**Comment by G1ov4** [comment]
I personally think that this:
![immagine](https://user-images.githubusercontent.com/33577286/93001660-ef491180-f530-11ea-9eb8-2232d892d8e5.png)
is way worse than this:
![immagine](https://user-images.githubusercontent.com/33577286/93001698-49e26d80-f531-11ea-9576-0e95a478b772.png)

**Comment by GenshinMT** [comment]
The only thing i may point out is that chat box must be toggle-able, since not everybody likes to see chat often in game

Some complains that it gets in the way.
